libraries/redcore/layouts/searchtools/default.php

```diff
@@ -23,7 +23,7 @@
 	'orderFieldSelector'  => '#list_fullordering'
 );
 
-$data['options'] = array_unique(array_merge($customOptions, $data['options']));
+$data['options'] = array_replace($customOptions, $data['options']);
```

Comment:
I double checked my tests and for 10k loops and found that I was using the same profiler in my previous comparison. Tests are now:
```
0.018 seconds (0.018); 8.87 MB (8.872) - array_merge
0.011 seconds (0.011); 8.87 MB (8.875) - array_replace
```

So :+1: here to array_replace

Joomla core has many bugs but we have to keep both synchronised as much as possible to avoid having 2 totally different systems for handling the same things. That's why contributing back is important.